### PR TITLE
[boost] Update to 1.72.0

### PR DESCRIPTION
**Pull request by yurybura**
Update Boost libarary to 1.72.0:
*  `generate-ports.ps1` script was modified to support new version and was run to re-generate related ports;
* `boost-python` was fixed to find Python3 even Python2 port is installed.
 
Fixes issue #9298 #6310

### CODE REVIEW

File: ports/boost-python/portfile.cmake [hunk 1]
```diff
@@ -5,13 +5,14 @@ include(vcpkg_common_functions)
...
-    REF boost-1.71.0
-    SHA512 0b78db2d0fda09298092cfbe7580d67db015aac7d5fac44c363dcc025f61d043fe4a87446b3f8f1fa76ae685838a98b0ed68609ecd945328f5fc4b12a493ee6a
+    REF boost-1.72.0
+    SHA512 9e2214f45b6b7e269caeba3decf2b79df9d1addc711aa65b75273e5fef9ff7423d400e4f11a904f932759b32a5c7e699441961494e061c8dd9c966da1f5b8fef
     HEAD_REF master
 )
 
 # Find Python. Can't use find_package here, but we already know where everything is
-file(GLOB PYTHON_INCLUDE_PATH "${CURRENT_INSTALLED_DIR}/include/python[0-9.]*")
+set(PYTHON_DIR_NAME "python3\.*")
```
Comment (cenit) [inline ports/boost-python/portfile.cmake]: This dir_name won't work on *nix. Please change the backslash to a forward slash

File: scripts/boost/post-source-stubs/python.cmake [hunk 2]
```diff
@@ -1,5 +1,6 @@
 # Find Python. Can't use find_package here, but we already know where everything is
-file(GLOB PYTHON_INCLUDE_PATH "${CURRENT_INSTALLED_DIR}/include/python[0-9.]*")
+set(PYTHON_DIR_NAME "python3\.*")
```
Comment (cenit) [inline scripts/boost/post-source-stubs/python.cmake]: Same here

**Review by cenit — COMMENTED** [review]
Please also edit the generate-ports.ps1 to remove `include(vcpkg_common_functions)`

File: ports/boost-python/portfile.cmake (hunk 1 above)
Comment (yurybura) [reply ports/boost-python/portfile.cmake]: `\.` matches the character `.` literally. Include path must be [...]/python3.6 or [...]/python3.7, etc.

File: scripts/boost/post-source-stubs/python.cmake (hunk 2 above)
Comment (yurybura) [reply scripts/boost/post-source-stubs/python.cmake]: Fixed.

File: ports/boost-python/portfile.cmake (hunk 1 above)
Comment (yurybura) [reply ports/boost-python/portfile.cmake]: Fixed.

**Comment by yurybura** [comment]
> This means that libgo is not buildable anymore in any os nor any config?

Temporary yes, this library is not compatible with Boost.Context 1.72.0.

**Review by PhoebeHui — APPROVED** [review]
Thanks for the PR!

File: scripts/boost/generate-ports.ps1 [hunk 3]
```diff
@@ -164,9 +154,6 @@ function Generate()
                 ")"
             )
         }
-        elseif ($Name -eq "iostreams")
```
Comment (ras0219-msft) [inline scripts/boost/generate-ports.ps1]: Please revert the changes to how `boost-iostreams` is handled and submit it as a separate PR

**Review by ras0219-msft — CHANGES_REQUESTED** [review]

**Comment by PhoebeHui** [comment]
Related to https://github.com/microsoft/vcpkg/pull/8372

File: ports/boost-system/portfile.cmake [hunk 4]
```diff
@@ -2,13 +2,11 @@
 
 include(vcpkg_common_functions)
 
-vcpkg_buildpath_length_warning(37)
```
[the remark this answers is not in the document]
Comment (yurybura) [reply ports/boost-system/portfile.cmake]: I've added it for boost-system only (like it was before).

File: scripts/boost/generate-ports.ps1 (hunk 3 above)
Comment (yurybura) [reply scripts/boost/generate-ports.ps1]: Reverted.

**Comment by yurybura** [comment]
Dear @dan-shaw, CI was failed with empty log,.. could you help to investigate the reason?

**Comment by dan-shaw** [comment]
Should work now

**Comment by yurybura** [comment]
Finally all checks have done. Please merge or request changes. Thanks a lot.

**Comment by vicroms** [comment]
Thanks for the PR!

**Pull request merged**